src/core/qfieldcloud/qfieldcloudproject.cpp

```diff
@@ -393,6 +420,36 @@ void QFieldCloudProject::refreshFileOutdatedStatus()
   } );
 }
 
+void QFieldCloudProject::downloadThumbnail()
+{
+  QgsLogger::debug( QStringLiteral( "Project %1: thumbnail download initiated." ).arg( mId ) );
+
+  if ( !mCloudConnection )
+    return;
+
+  QNetworkRequest request;
+  request.setAttribute( QNetworkRequest::RedirectPolicyAttribute, QNetworkRequest::RedirectPolicy::NoLessSafeRedirectPolicy );
+  mCloudConnection->setAuthenticationDetails( request );
+
+  NetworkReply *reply = mCloudConnection->get( request, QStringLiteral( "/api/v1/files/thumbnails/%1/" ).arg( mId ) );
+  connect( reply, &NetworkReply::finished, reply, [=]() {
+    QNetworkReply *rawReply = reply->currentRawReply();
+
+    Q_ASSERT( reply->isFinished() );
+    Q_ASSERT( reply );
+
+    if ( rawReply->error() == QNetworkReply::NoError )
+    {
+      QTemporaryFile file( QString( "%1/XXXXXX.%2" ).arg( QDir::tempPath(), QStringLiteral( "PNG" ) ) );
```

Comment:
Is it required to hardcode the file extension? You can easily extract it with `QFileInfo`.

Reply:
@suricactus , we don't have a filename for project thumbnails, only a content-[mime]type. In any case I added a bit of logic to get the image extension through content-type when available.